src/scenes/level_scene.py

```diff
@@ -2126,6 +2131,15 @@ def right_click(self) -> None:
                                     self.selected_player.remove_item(item[0])
                                     item[1].set_item(item[0])
                             self.traded_items.clear()
+                        if self.traded_gold:
+                            # Return traded gold
+                            for gold in self.traded_gold:
+                                if gold[1] == self.selected_player:
+                                    self.selected_player.gold += gold[0]
+                                    gold[2].gold -= gold[0]
+                                else:
+                                    self.selected_player.gold -= gold[0]
+                                    gold[2].gold += gold[0]
```

Comment:
You are missing the line to clear the `traded_gold` collection.

As a consequence, it's possible to cancel the turn of the same character again and again and the gold is piling up for one character, and the other one starts to have big debts 🥲

Reply:
Thank you for your careful review, I am very sorry for making this low-level mistake.

Reply:
I'm also making this kind of mistakes on daily basis, don't worry about it.